be/src/vec/runtime/vdatetime_value.cpp

```diff
@@ -1580,10 +1580,10 @@ bool VecDateTimeValue::date_add_interval(const TimeInterval& interval) {
         // This will change month and year information, maybe date.
         int64_t months = _year * 12 + _month - 1 + sign * (12 * interval.year + interval.month);
         _year = months / 12;
-        if (_year > 9999) {
+        if (_year < MIN_YEAR || _year > MAX_YEAR) {
             return false;
         }
-        _month = (months % 12) + 1;
+        _month = ((_month + 11 + sign * interval.month + 12) % 12) + 1;
```

Comment:
I see a error about `runtime error: index 252 out of bounds for type 'int [13]'`
the change could cover that error ?

Reply:
Is there a error case for me to test？

Reply:
now update.
In my opinion, if months is less than 0, it should be an invalid value.If we consider negative numbers as valid values, then the result of months = 5 and months = -7 will be the same.